yt_dlp/YoutubeDL.py

```diff
@@ -3494,9 +3509,22 @@ def download_with_info_file(self, info_filename):
             # FileInput doesn't have a read method, we can't call json.load
             infos = [self.sanitize_info(info, self.params.get('clean_infojson', True))
                      for info in variadic(json.loads('\n'.join(f)))]
+
         for info in infos:
-            self._load_cookies(info.get('cookies'), from_headers=False)
-            self._load_cookies(traverse_obj(info.get('http_headers'), 'Cookie', casesense=False))  # compat
+            if info.get('cookies'):
+                self._load_cookies(info['cookies'], autoscope=False)
+            else:  # compat: Auto-scope header cookies, add to cookiejar & purge from http_headers
+                self._load_cookies(traverse_obj(
+                    info.get('http_headers'), 'Cookie', casesense=False), autoscope=info['url'])
+            info['http_headers'] = self._remove_cookie_header(info.get('http_headers')) or None
+            for fmt in info.get('formats') or []:
+                if fmt.get('cookies'):
+                    self._load_cookies(fmt['cookies'], autoscope=False)
+                else:  # compat
+                    self._load_cookies(traverse_obj(
+                        fmt.get('http_headers'), 'Cookie', casesense=False), autoscope=fmt['url'])
+                fmt['http_headers'] = self._remove_cookie_header(fmt.get('http_headers')) or None
```

Comment:
info can be of playlist, url_reult etc. Do this in `process_video_result`

Reply:
Ended up keeping the `info['http_headers']` filtering inside `download_with_info_file` (or else I think subtitle downloads etc could grab the old infojson's header cookies), but otherwise moved everything into `process_video_result`